docs/source/wrappers.rst

```diff
@@ -0,0 +1,499 @@
+                # perform some transformation on only one part of the input
+                self.input_encoder = OneHotEncoder()
+                X[1] = self.input_encoder.fit_transform(X[1])
+                return X, extra_args
+
+    clf = MyWrapper(
+        model=model_build_fn,
+        loss=["binary_crossentropy", "categorical_crossentropy"],
+        optimizer="adam"
+    )
+    clf.fit(X, y)
+
+
+.. _param-routing:
+
+Routed parameters
+-----------------
+
+For more advanced used cases, SciKeras supports
+Scikit-Learn style parameter routing to override parameters
+for individual consumers (methods or class initializers).
+
+All special prefixes are stored in the ``prefixes_`` class attribute
+of :py:class:`scikeras.wrappers.BaseWrappers`. Currently, they are:
+
+- ``model__``: passed to ``model_build_fn`` (or whatever function is passed to the ``model`` param of :class:`scikeras.wrappers.BaseWrapper`).
+- ``fit__``: passed to :func:`tensorflow.keras.Model.fit`
+- ``predict__``: passed to :func:`tensorflow.keras.Model.predict`. Note that internally SciKeras also uses :func:`tensorflow.keras.Model.predict` within :func:`scikeras.wrappers.BaseWrapper.score` and so this prefix applies to both.
+- ``callbacks__``: used to instantiate callbacks.
+- ``optimizer__``: used to instantiate optimizers.
+- ``loss__``: used to instantiate losses.
+- ``metrics__``: used to instantiate metrics.
+- ``score__``: passed to the scoring function, i.e. :func:`scikeras.wrappers.BaseWrapper.scorer`.
+
+All routed parameters will be available for hyperparameter tuning.
+
+Below are some example use cases.
+
+Example: multiple losses with routed parameters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code:: python
+
+    from tensorflow.keras.losses import BinaryCrossentropy, CategoricalCrossentropy
+
+    clf = KerasClassifier(
+        model=model_build_fn,
+        loss=[BinaryCrossentropy, CategoricalCrossentropy],
+        loss__from_logits=True,  # BinaryCrossentropy(from_logits=True) & CategoricalCrossentropy(from_logits=True)
+        loss__label_smoothing=0.1,  # passed to each sub-item, i.e. `loss=[l(label_smoothing=0.1) for l in loss]`
+        loss__1__label_smoothing=0.5,  # overrides the above, results in CategoricalCrossentropy(label_smoothing=0.5)
+    )
+
+To use this syntax, you must pass classes instead of strings or instances.
+For even more complex use cases, you can use the special suffix ``__param_groups``
+and pass a list of ``("destination", param_dict)`` instead of individual parameters. This also allows use of
+regex-based keys.
+
+.. code:: python
+
+    from tensorflow.keras.losses import BinaryCrossentropy, CategoricalCrossentropy
+
+    clf = KerasClassifier(
+        model=model_build_fn,
+        loss={
+            "binary_output_1": BinaryCrossentropy,
+            "binary_output_2": BinaryCrossentropy,
+            "categorical_output_1: CategoricalCrossentropy,
+        }
+        loss__param_groups=[
```

Comment:
Has a `param_group` syntax been implemented?

Reply:
Not yet. I think it shouldn't be too hard to do once #66 is merged.